compiler/rustc_type_ir/src/lib.rs

```diff
@@ -251,6 +252,9 @@ bitflags! {
 
         /// Does this value have `InferConst::Fresh`?
         const HAS_CT_FRESH                = 1 << 19;
+
+        /// Does this have `TyAlias`?
+        const HAS_TY_ALIAS                = 1 << 20;
```

Comment:
I think you need to add this flag to `needs_normalization` in `rustc_trait_selection::traits::project` as well

Reply:
Indeed, good catch!